cloud/pkg/upgradecontroller/controller/upstream.go

```diff
@@ -0,0 +1,223 @@
+	"strings"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apimachineryType "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	beehiveContext "github.com/kubeedge/beehive/pkg/core/context"
+	"github.com/kubeedge/beehive/pkg/core/model"
+	keclient "github.com/kubeedge/kubeedge/cloud/pkg/common/client"
+	"github.com/kubeedge/kubeedge/cloud/pkg/common/messagelayer"
+	"github.com/kubeedge/kubeedge/cloud/pkg/upgradecontroller/config"
+	"github.com/kubeedge/kubeedge/common/types"
+	"github.com/kubeedge/kubeedge/pkg/apis/upgrade/v1alpha2"
+	crdClientset "github.com/kubeedge/kubeedge/pkg/client/clientset/versioned"
+)
+
+// UpstreamController subscribe messages from edge and sync to k8s api server
+type UpstreamController struct {
+	kubeClient   kubernetes.Interface
+	crdClient    crdClientset.Interface
+	messageLayer messagelayer.MessageLayer
+	// message channel
+	upgradeStatusChan chan model.Message
+
+	// downstream controller to update Upgrade status in cache
+	dc *DownstreamController
+}
+
+// Start UpstreamController
+func (uc *UpstreamController) Start() error {
+	klog.Info("Start Upgrade Upstream Controller")
+
+	uc.upgradeStatusChan = make(chan model.Message, config.Config.Buffer.UpdateUpgradeStatus)
+	go uc.dispatchMessage()
+
+	for i := 0; i < int(config.Config.Load.UpgradeWorkers); i++ {
+		go uc.updateUpgradeStatus()
+	}
+	return nil
+}
+
+// Start UpstreamController
+func (uc *UpstreamController) dispatchMessage() {
+	for {
+		select {
+		case <-beehiveContext.Done():
+			klog.Info("Stop dispatch upgrade upstream message")
+			return
+		default:
+		}
+
+		msg, err := uc.messageLayer.Receive()
+		if err != nil {
+			klog.Warningf("Receive message failed, %s", err)
+			continue
+		}
+
+		klog.V(4).Infof("Upgrade upstream controller receive msg %#v", msg)
+
+		uc.upgradeStatusChan <- msg
+	}
+}
+
+// Start UpstreamController
+func (uc *UpstreamController) updateUpgradeStatus() {
+	for {
+		select {
+		case <-beehiveContext.Done():
+			klog.Info("Stop updateUpgradeStatus")
+			return
+		case msg := <-uc.upgradeStatusChan:
+			klog.V(4).Infof("Message: %s, operation is: %s, and resource is: %s", msg.GetID(), msg.GetOperation(), msg.GetResource())
+
+			// get nodeID and upgradeID from Upgrade msg:
+			nodeID := getNodeName(msg.GetResource())
+			upgradeID := getUpgradeID(msg.GetResource())
+
+			oldValue, ok := uc.dc.upgradeManager.UpgradeMap.Load(upgradeID)
+			if !ok {
+				klog.Errorf("Upgrade %s not exist", upgradeID)
+				continue
+			}
+
+			nodeInfo, err := uc.kubeClient.CoreV1().Nodes().Get(context.Background(), nodeID, metav1.GetOptions{})
+			if err != nil {
+				klog.Errorf("Failed to get node info: %v", err)
+				continue
+			}
+
+			// mark edge node schedulable
+			// the effect is like running cmd: kubectl uncordon <node-to-drain>
+			if nodeInfo.Labels != nil && nodeInfo.Labels["upgrade"] == "upgrade" {
+				nodeInfo.Spec.Unschedulable = false
```

Comment:
Suggest the key and the value be different.  How about `upgrade.kubeedge.io/status=true`?  :)

Reply:
how about `upgrade.kubeedge.io/upgrade=""` 
i think this is enough, just like edge-node label `"node-role.kubernetes.io/edge": ""`

Reply:
edge subordinates to of "node-role",  we need a key to subordinate upgrade steps.  I vote for `upgrade.kubeedge.io/status=“”`, but it doesn't mean to disagree with you.